### DISCUSSION TITLE
What is the size of the package that gets downloaded to the client?

### DISCUSSION THREAD

**Discussion by bengrunfeld** [prisma/prisma#19255] [Q&A]
On the [Prisma Client npm page](https://www.npmjs.com/package/@prisma/client), it says that the unpacked size of Prisma client is 3.31 MB. Does that mean that the client has to download 3.31 MB? If not, how big is the package that ultimately ends up in the client?

**Comment by ludralph** [comment]
Hi @bengrunfeld 👋 

You are correct that the client has to download 3.31MB.

**Reply by bengrunfeld** [reply]
@ludralph isn't that prohibitively big? Especially for mobile device / low bandwidth? So it Prisma only really appropriate for web apps where the user is expected to have a fast internet connection?

**Comment by janpio** [comment]
That is a more complex question than you assumed @bengrunfeld I think :D Let me give you an overview:

The npm page shows the size of the _unpacked_ npm package, currently 3.31 MB, so after it has been downloaded as an archive and then extracted by your package manager, so the size it will take on your disk. The actual download size is hence smaller, and for example, this tool lists it: https://arve0.github.io/npm-download-size/#@prisma%2fclient, <1MB for @prisma/client@4.14.0`.

But `@prisma/client` only works together with Prisma CLI, which is the `prisma` npm package. That is also needed in your project, so depending on what you are looking for, you might want to add its values to the first ones.

When you install the CLI package, and when you `prisma generate` your own _custom_ Client adapted to your Prisma schema, the Prisma CLI will download the Prisma engines for _your_ operating system (and configuration). The size of these engine files is different per operating system. The best way to figure out those numbers is really just to do it and then look at your `node_modules` and the files put there.

And then of course at the end all this can be mixed up by your app or framework, if it uses a bundle or minifier for your code. The CLI _and_ the `@prisma/client` package both might not even end up in your app - only the generated Client (including one of the OS dependent engines) and some runtimes from `@prisma/client` are really needed for the generated Client to work.

Does this include the information you were looking for?

**Reply by nurul3101** [reply]
@bengrunfeld Did you had a chance to look at Jan's response?

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
